### PR TITLE
Fixed the failing splunkforwarder test

### PR DESCRIPTION
**Pull request by aliceh**

### CODE REVIEW

File: pkg/e2e/operators/splunkforwarder.go [hunk 1]
```diff
@@ -163,7 +164,12 @@ func dedicatedAaddSplunkforwarder(SplunkForwarder sfv1alpha1.SplunkForwarder, na
 		return err
 	}
 	unstructuredObj := unstructured.Unstructured{obj}
-	h.SetServiceAccount("system:serviceaccount:%s:dedicated-admin-project")
+	h.Impersonate(rest.ImpersonationConfig{
+		UserName: "test-user@redhat.com",
+		Groups: []string{
+			"dedicated-admins",
+		},
+	})
```
Comment (mrbarge) [inline pkg/e2e/operators/splunkforwarder.go:172]: I can't see anywhere where it's resetting the impersonation when the test finishes. There should likely be something like the following, in keeping with the convention that other tests do.

```
      defer func() {
        h.Impersonate(rest.ImpersonationConfig{})                                                                                                                                                               
      }()
```

Comment (aliceh) [reply pkg/e2e/operators/splunkforwarder.go:172]: Added the reset step. Thanks for reviewing, @mrbarge !

**Comment by mrbarge** [comment]
Verified test succeeded on a staging cluster.
/lgtm
/approve

**Pull request merged**